### PR TITLE
docs: add readme

### PR DESCRIPTION
**Pull request by GCdePaula**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,56 @@
...
+
+
+## User transactions
+
+Users build an [EIP-712](https://eips.ethereum.org/EIPS/eip-712) signed transaction. A `SignedTransaction` consists of the pair `SigningMessage` and a `Signature`. They then submit it to a sequencer frontend.
+
+Note that this signed transaction contains the address of the destination dapp. It also contains a nonce (which is per application), and the maximum gas price this user is willing to pay _for DA_.
+
+
+## Sequencer frontend
```
Comment (augustoteixeira) [inline README.md:20]: I don't know if I like the name "frontend"... but I can't come up with a better one.

File: README.md [hunk 2]
```diff
@@ -1 +1,56 @@
...
+
+There are many kinds of sequencer front end:
+* In a centralized sequencer, users send transactions to the sequencer through a “submit endpoint”.
+* In a based sequencer, users send transactions to a mempool/p2p-network, and the set of Ethereum block builders running some version of chorizo will pick these transactions up.
+* In Espresso, users send transactions to the Espresso network, which is then picked up by the sequencers, and finally the elected builder has the rights to sequence these transactions.
+
+
+## Sequencer batcher
+
+The batcher is the component in the sequencer that takes an unordered set of `SignedTransaction`s, and builds a `Batch`, consisting of an ordered set of `WireTransaction`s.
```
Comment (augustoteixeira) [inline README.md]: Calling the transactions "unordered" is perhaps too much. It is just that other people may disagree on their exact order. But the sequencer received one after another.

File: README.md [hunk 3]
```diff
@@ -1 +1,56 @@
...
+Here, the sequencer has the freedom to do anything (except forge signatures, since it’s cryptographically impossible).
+They can reorder transactions, compress them, aggregate signatures, whatever.
```
Comment (augustoteixeira) [inline README.md]: Ideally they should not reorder :)

File: README.md [hunk 4]
```diff
@@ -1 +1,56 @@
...
+As an important detail, a `Batch` contains a single “payment address”, which we tacitly assume is the wallet of the sequencer who created the batch.
+Also, note that a `Batch` contains transactions with different destination apps/addresses.
+
+The important part is that, by looking just at a `Batch` and perhaps some context, a `Batch` can be parsed and verified into a list of `Transaction`s.
+The simplest batcher orders the user signed transactions in a first come first serve policy, and just appends and serializes all transactions into a blob.
+
+
+## Sequencer backend
+
+The backend is the component in the sequencer that takes a `Batch` and submits it to a DA layer. It could in Ethereum calldata, Ethereum 4844 blob, Espresso DA, Avail, Celestia, etc.
```
Comment (augustoteixeira) [inline README.md]: It could <be> in Ethereum...

File: README.md [hunk 5]
```diff
@@ -1 +1,56 @@
...
+
+
+## Payment app
+
+One of these apps is special: it is the payment app.
+This app is written by us, we run a validator to protect it, and has our seal of approval. Sequencers must trust this app, but not any of the others.
+This app also has a batch parser, and a wallet.
+
+Users will deposit money on this app, which will pay for DA costs incurred by the sequencer.
+After parsing a batch, this app will transfer ether from each user that submitted a transaction to the sequencer wallet (i.e. “payment address”), using the maximum gas price signed by the user, and the size of the transaction payload.
```
Comment (augustoteixeira) [inline README.md]: Could we use the real gas-cost coming from the base layer as metadata and only truncate it if it surpasses max-gas?

**Comment by augustoteixeira** [comment]
I did some small comments. You can address the ones that you think make sense and merge.

**Review by augustoteixeira — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1 +1,56 @@
 # cartesi-paio
```
Comment (miltonjonat) [inline README.md]: I think we should have an introduction section explaining the motivations and main concepts. Something along the lines of the potential for lower costs when batching tx's, coupled with having better convenience/UX when using alt-DA.
Then, it should explain the main parts (frontend, backend, batcher, etc), preferably with a diagram. It should also cover the main expected workflow: e.g., user sends tx to the sequencer, which batches them (as long as they pay for it) and sends the batched tx blob to some DA; then, both a wallet dapp and the destination dapps process the blobs, with the wallet paying the sequencer for its work while the dapp processes the blob transactions

File: README.md [hunk 7]
```diff
@@ -1 +1,56 @@
...
+
+
+## User transactions
+
+Users build an [EIP-712](https://eips.ethereum.org/EIPS/eip-712) signed transaction. A `SignedTransaction` consists of the pair `SigningMessage` and a `Signature`. They then submit it to a sequencer frontend.
```
Comment (miltonjonat) [inline README.md]: `SignedTransaction` has a `SigningMessage`? Unsure why you use "signing" here. Should be "SignedMessage" or maybe just "Message"?

File: README.md (hunk 1 above)
Comment (miltonjonat) [reply README.md:20]: I agree frontend sounds weird here. It feels like there would be a CLI or a UI, and it has nothing to do with that. Maybe something like "submission API"?

File: README.md [hunk 8]
```diff
@@ -1 +1,56 @@
...
+## Sequencer batcher
+
+The batcher is the component in the sequencer that takes a list of `SignedTransaction`s and builds a `Batch`, consisting of an ordered set of `WireTransaction`s.
+Here, the sequencer has the freedom to do anything (except forge signatures, since it’s cryptographically impossible).
+They can compress transactions, aggregate signatures, and even reorder transactions (although they ideally shouldn't).
+
+As an important detail, a `Batch` contains a single “payment address”, which we tacitly assume is the wallet of the sequencer who created the batch.
+Also, note that a `Batch` contains transactions with different destination apps/addresses.
+
+The important part is that, by looking just at a `Batch` and perhaps some context, a `Batch` can be parsed and verified into a list of `Transaction`s.
```
Comment (miltonjonat) [inline README.md]:
```suggestion
The important part is that, by looking just at a `Batch` and perhaps some context, a `Batch` can be parsed and verified into a list of `Transaction`s.
```
The important part is that a `Batch` can be parsed and verified into a list of `Transaction`s just by looking at it (perhaps along with some context)

File: README.md [hunk 9]
```diff
@@ -1 +1,56 @@
...
+The simplest batcher orders the user signed transactions in a first come first serve policy, and just appends and serializes all transactions into a blob.
+
+
+## Sequencer backend
+
+The backend is the component in the sequencer that takes a `Batch` and submits it to a DA layer, such as Ethereum calldata, Ethereum 4844, Espresso DA, Avail, Celestia, etc.
```
Comment (miltonjonat) [inline README.md]:
```suggestion
The backend is the component in the sequencer that takes a `Batch` and submits it to a DA layer, such as Ethereum calldata, Ethereum 4844, Espresso DA, Avail DA, Celestia, etc.
```

**Review by miltonjonat — APPROVED** [review]

**Pull request merged**